src/components/split_button/split_button_control.tsx

```diff
@@ -0,0 +1,122 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import React, {
+  FunctionComponent,
+  ButtonHTMLAttributes,
+  ReactNode,
+} from 'react';
+// import classNames from 'classnames';
+
+import { CommonProps } from '../common';
+
+// import { htmlIdGenerator } from '../../services/accessibility';
+// import { OuiFormControlLayoutProps } from '../form/form_control_layout';
+// import { OuiFlexGroup, OuiFlexItem } from '../flex';
+import {
+  ButtonColor,
+  ButtonSize,
+  OuiButton,
+  OuiButtonIcon,
+  OuiButtonIconColor,
+} from '../button';
+
+export type OuiSplitButtonOption = string | ReactNode;
+
+export interface OuiSplitButtonControlProps
+  extends CommonProps,
+    ButtonHTMLAttributes<HTMLButtonElement> {
+  fullWidth?: boolean;
+  isLoading?: boolean;
+
+  fill?: boolean;
+
+  /**
+   * Any of our named colors.
+   */
+  color?: ButtonColor;
+
+  /**
+   * Use size `s` in confined spaces
+   */
+  size?: ButtonSize;
+
+  /**
+   * Click handler of Primary button
+   */
+  onClick?: () => void;
+
+  /**
+   * Click handler for drop-down button
+   */
+  onDropdownClick?: () => void;
+
+  /**
+   * Content of Primary (left-side) button
+   */
+  children?: ReactNode;
+}
+
+export const OuiSplitButtonControl = ({
+  fill,
+  size,
+  color,
+  children,
+  fullWidth = false,
+  onClick,
+  onDropdownClick,
+}: OuiSplitButtonControlProps): ReturnType<
+  FunctionComponent<OuiSplitButtonControlProps>
+> => {
+  // const classes = classNames(
+  //   'ouiSplitButtonControl',
+  //   {
+  //     'ouiSplitButtonControl--fullWidth': fullWidth,
+  //     'ouiSplitButtonControl--compressed': compressed,
+  //     'ouiSplitButtonControl-isLoading': isLoading,
+  //     'ouiSplitButtonControl-isInvalid': isInvalid,
+  //   },
+  //   className
+  // );
+
+  // const icon: OuiFormControlLayoutProps['icon'] = {
+  //   type: 'arrowDown',
+  //   side: 'right',
+  // };
+
+  const iconDisplay = fill ? 'fill' : 'base';
+
+  // const screenReaderId = htmlIdGenerator()();
+
+  return (
+    <div className="ouiSplitButtonControl">
+      <OuiButton
+        className="ouiSplitButtonControl--primary"
+        fill={fill}
+        color={color}
+        size={size}
+        fullWidth={fullWidth}
+        onClick={onClick}>
+        {children}
+      </OuiButton>
+      <OuiButtonIcon
+        className="ouiSplitButtonControl--dropdown"
+        display={iconDisplay}
+        color={color as OuiButtonIconColor}
+        size={size || 'm'}
```

Comment:
There's a type warning here... don't know what the right mix of ButotnColor into OuiButtonIconColor.

Reply:
Yeah, there is probably some difference between `ButtonColor` and `OuiButtonIconColor` types. I would recommend just using `OuiButtonIconColor` in the props type to begin with so you don't need to worry about type coercion like this

Reply:
I've tried both types `ButtonColor` or `OuiButtonIconColor`.  Union won't give a clear typedef when put on either component.
I tried several other union/exclusion combinations.
I've even tried a custom type that is a bespoke intersection of the two...
and still it warns on type matching for the <OuiButtonIcon color= 

For the moment I'm going to commit the bespoke type and maybe someone will figure out how to manipulate this confilcting typedef.